### ISSUE TITLE
Docker Container: module has no attribute

### ISSUE THREAD

**Issue by chrisgerdener** [AMWA-TV/nmos-testing#821]
Pretty much as the title says. Tried to use the test tool via docker container. When we try to run the IS-04 Node API test suite, we get an error stating: ERROR: module 'asyncio' has no attribute 'get_running_loop'
Happens on the latest docker image from June 28

**Comment by garethsb** [comment]
Hmm, strange... [`asyncio.get_running_loop`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop) has been around since 3.7. Will investigate.

**Comment by garethsb** [comment]
Are you building the container locally from the [Dockerfile](https://github.com/AMWA-TV/nmos-testing/blob/12f1b053da339606eb98c67dae8106edde585e9f/Dockerfile), or using the pre-built [container image published at Docker Hub](https://hub.docker.com/layers/amwa/nmos-testing/master-aa58b5e/images/sha256-bb7841f89e66d097283551135020a8734164826def34689581c473f6cdb6924a?context=explore)?

**Comment by garethsb** [comment]
> Hmm, strange... [`asyncio.get_running_loop`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop) has been around since 3.7. Will investigate.

Oh. Well. That's probably it. Dockerfile says `FROM ubuntu:bionic` which is 18.04, which shipped with Python 3.6. And that is indeed our claimed oldest supported version in [Dependencies](https://github.com/AMWA-TV/nmos-testing/blob/aa58b5ed32e8252249b3345c9b58fb66943cbd9d/docs/1.1.%20Installation%20-%20Local.md#dependencies).

Since we don't directly call `asyncio.get_running_loop` I assume one of our unpinned dependencies in [_requirements.txt_](https://github.com/AMWA-TV/nmos-testing/blob/aa58b5ed32e8252249b3345c9b58fb66943cbd9d/requirements.txt) does, which means it inadvertently bumped our minimum requirement.

I'd like to find out which... but also not against bumping our minimum Python requirement and building the Docker image `FROM ubuntu:focal` (20.04). [Bionic (18.04) is end of life already.](https://ubuntu.com/18-04)

**Comment by chrisgerdener** [comment]
> Are you building the container locally from the [Dockerfile](https://github.com/AMWA-TV/nmos-testing/blob/12f1b053da339606eb98c67dae8106edde585e9f/Dockerfile), or using the pre-built [container image published at Docker Hub](https://hub.docker.com/layers/amwa/nmos-testing/master-aa58b5e/images/sha256-bb7841f89e66d097283551135020a8734164826def34689581c473f6cdb6924a?context=explore)?

We pulled the latest prebuilt image from docker  hub according to the install instructions.

**Comment by duncwebb** [comment]
> > Hmm, strange... [`asyncio.get_running_loop`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop) has been around since 3.7. Will investigate.
> 
> Oh. Well. That's probably it. Dockerfile says `FROM ubuntu:bionic` which is 18.04, which shipped with Python 3.6. And that is indeed our claimed oldest supported version in [Dependencies](https://github.com/AMWA-TV/nmos-testing/blob/aa58b5ed32e8252249b3345c9b58fb66943cbd9d/docs/1.1.%20Installation%20-%20Local.md#dependencies).
> 
> Since we don't directly call `asyncio.get_running_loop` I assume one of our unpinned dependencies in [_requirements.txt_](https://github.com/AMWA-TV/nmos-testing/blob/aa58b5ed32e8252249b3345c9b58fb66943cbd9d/requirements.txt) does, which means it inadvertently bumped our minimum requirement.
> 
> I'd like to find out which... but also not against bumping our minimum Python requirement and building the Docker image `FROM ubuntu:focal` (20.04). [Bionic (18.04) is end of life already.](https://ubuntu.com/18-04)

Looking in the container I think it's zeroconf that's causing the problem (the 0.38.4 revision it has installed claims a requirement on CPython 3.7+). When it was pulled in as dependency of zeroconf-monkey it was at 0.28.5 which didn't make any calls to asyncio.get_running_loop (0.32.0 was the first to use this). I think claim of CPython 3.6+ in the README.md in 0.32.0 is a mistake. Python 3.6 has been EoL for over a year now - is there a requirement to support it still?

**Comment by garethsb** [comment]
I agree (see also #830). I'll bump minimum python to 3.7 as part of #829.
